Cannot static link to cpprestsdk 2.9

The vcpkg static build of the cpprestsdk is emitting a number of __declspec(dllimport) statements.

`1>test_site_json.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl web::uri::uri(wchar_t const *)" (__imp_??0uri@web@@QEAA@PEB_W@Z) referenced in function "void __cdecl core_test::____C_A_T_C_H____T_E_S_T____0(void)" (?____C_A_T_C_H____T_E_S_T____0@core_test@@YAXXZ)
1>test_site_json.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: __cdecl web::uri::uri(class std::basic_string<wchar_t,struct std::char_traits<wchar_t>,class std::allocator<wchar_t> > const &)" (__imp_??0uri@web@@QEAA@AEBV?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@@Z) referenced in function "void __cdecl core_test::____C_A_T_C_H____T_E_S_T____0(void)" (?____C_A_T_C_H____T_E_S_T____0@core_test@@YAXXZ)
1>test_site_json.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: bool __cdecl web::uri::operator==(class web::uri const &)const " (__imp_??8uri@web@@QEBA_NAEBV01@@Z) referenced in function "public: static bool __cdecl Catch::Internal::Evaluator<class web::uri,class web::uri,0>::evaluate(class web::uri const &,class web::uri const &)" (?evaluate@?$Evaluator@Vuri@web@@V12@$0A@@Internal@Catch@@SA_NAEBVuri@web@@0@Z)
`

This seems to be related to [https://github.com/Microsoft/cpprestsdk/issues/344](https://github.com/Microsoft/cpprestsdk/issues/344) 

regards

--- a/ports/cpprestsdk/CONTROL
+++ b/ports/cpprestsdk/CONTROL
@@ -1,5 +1,5 @@
 Source: cpprestsdk
-Version: 2.10.16-2
+Version: 2.10.16-3
 Build-Depends: openssl (!uwp&!windows), boost-system (!uwp&!windows),
   boost-date-time (!uwp&!windows), boost-regex (!uwp&!windows), boost-thread (!uwp&!windows),
   boost-filesystem (!uwp&!windows), boost-random (!uwp&!windows), boost-chrono (!uwp&!windows),

--- a/ports/cpprestsdk/portfile.cmake
+++ b/ports/cpprestsdk/portfile.cmake
@@ -45,4 +45,9 @@ vcpkg_copy_pdbs()
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/share/${PORT})
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib/share ${CURRENT_PACKAGES_DIR}/lib/share)
 
+if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/include/cpprest/details/cpprest_compat.h
+        "#ifdef _NO_ASYNCRTIMP" "#if 1")
+endif()
+
 file(INSTALL ${SOURCE_PATH}/license.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

